### PR TITLE
[Catalog] Expose colorScheme + typographyScheme properties for all examples to use app wide theming

### PR DESCRIPTION
**Pull request by yarneo**
* Made colorScheme and typographyScheme properties in the class interfaces for all our examples.
* Moved theming away from example inits and into viewDidLoad
* Small update for slider example to allow it to grab the theming properly.

Related Pivotal: https://www.pivotaltracker.com/story/show/156616695

### CODE REVIEW

File: components/AppBar/examples/AppBarDelegateForwardingExample.m [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
 @interface AppBarDelegateForwardingExample : UITableViewController
 @property(nonatomic, strong) MDCAppBar *appBar;
+@property(nonatomic, strong) MDCSemanticColorScheme *colorScheme;
```
Comment (jverkoey) [inline components/AppBar/examples/AppBarDelegateForwardingExample.m:26]: I know it's a pain in the butt in Objective-C, but we probably need to follow-up with an initialization of these properties to default values on each example.

Comment (jverkoey) [reply components/AppBar/examples/AppBarDelegateForwardingExample.m:26]: Please file a story to do as a follow-up and put at the bottom of the polish epic. Is not particularly critical.

File: components/NavigationBar/examples/NavigationBarTypicalUseExample.swift [hunk 2]
```diff
@@ -32,8 +31,7 @@ open class NavigationBarTypicalUseSwiftExample: NavigationBarTypicalUseExample {
     let mutator = MDCNavigationBarTextColorAccessibilityMutator()
     mutator.mutate(navBar!)
 
-    let colorScheme = MDCSemanticColorScheme()
-    MDCNavigationBarColorThemer.applySemanticColorScheme(colorScheme, to: navBar!)
+    MDCNavigationBarColorThemer.applySemanticColorScheme(colorScheme!, to: navBar!)
```
Comment (jverkoey) [inline components/NavigationBar/examples/NavigationBarTypicalUseExample.swift:34]: Why does this one need `!`?

**Review by jverkoey — APPROVED** [review]

File: components/AppBar/examples/AppBarDelegateForwardingExample.m (hunk 1 above)
Comment (yarneo) [reply components/AppBar/examples/AppBarDelegateForwardingExample.m:26]: naww >_< https://www.pivotaltracker.com/story/show/157064672

File: components/NavigationBar/examples/NavigationBarTypicalUseExample.swift (hunk 2 above)
Comment (yarneo) [reply components/NavigationBar/examples/NavigationBarTypicalUseExample.swift:34]: The interface itself is in an obj-c file

File: components/AppBar/examples/AppBarDelegateForwardingExample.m (hunk 1 above)
Comment (jverkoey) [reply components/AppBar/examples/AppBarDelegateForwardingExample.m:26]: yaaawwww >.<

File: components/NavigationBar/examples/NavigationBarTypicalUseExample.swift (hunk 2 above)
Comment (jverkoey) [reply components/NavigationBar/examples/NavigationBarTypicalUseExample.swift:34]: Ah got it.

**Pull request merged**